File: KwangHo/H-Index.py
def solution(citations):
    citations.sort() #내장함수 정렬
    answer = 0
    for i in range(0,len(citations)): # 0~리스트길이
        if len(citations)-i <= citations[i] : #인용 최소화 기준으로 맞쳐야 h번 인용이 딱 맞아 떨어진다.
            answer = len(citations)-i         #그래서 현재부터~끝까지의 길이와 인용횟수를 비교한다.
            break                             #길이를 넣어주고 나오면 끝!!
        
    return answer

'''
이문제는 H-Index이해하는데 한참걸렸다.(생소)
요점은
1. h번 인용 h번
2. 그리고 이 h번 인용이 최소화
문제이해한 후로는 잘 풀렸다.
'''

File: Doodream/H-index.js
function solution(citations) {

    //https://programmers.co.kr/learn/courses/30/lessons/42747?language=javascript

    let arrH = [];

    // h를 0부터 차례 차례 증가시킨다.
    for (let h = 0; h <= citations.length; h++) {
        let count = 0;
        for (let i = 0; i < citations.length; i++) {
            // h번이상 인용되었다면 count++
            if (citations[i] >= h) {
                count++
            }
        }
        // h번이상 인용된 논문의 수가 h이상이면 해당 h를 arrH에 추가
        if (count >= h) {
            arrH.push(h);
        } else {
            break;
        }
    }

    // 작은순으로 추가되므로 가장 마지막에 있는 원소가 가장 큰 H
    var answer = arrH[arrH.length - 1];
    return answer;
}

let citations = [0, 0, 0];
console.log(solution(citations));